modules/protect.php

```diff
@@ -858,15 +856,15 @@ function get_transient( $transient ) {
 		return get_transient( $transient );
 	}
 
+	/**
+	 * Get the API host.
+	 *
+	 * @return string
+	 *
+	 * @deprecated 9.0.0 Use constant `JETPACK_PROTECT__API_HOST` instead.
+	 */
 	function get_api_host() {
```

Comment:
I agree that if we're going to deprecate it, we should fully deprecate it -- remove all uses and use the `_deprecated_function()` call unless there is a particular reason for not doing so. For example, in the past, we delayed calling `_deprecated_function` when Akismet and VaultPress called a function that we were deprecating to give them time to update it. It was a known issue and they were the only known external users of that function, so there was no benefit to throw the deprecation notice into the PHP logs.

We tend to leave code in a bit longer—we're just now looking at removing things that have been deprecated for six months or more. With all of the potential ways that plugins/themes could be calling on some of the code (and depending on how long it has existed), we're a been conservative on actually removing deprecated code—especially if it basically just calls a new supported function.

Reply:
@anomiex, @kraftbj 
You're right, I cut some corners to make the testing a little bit easier.
But since it's already unsafe to merge into `9.0`, I updated all those features and we'll merge it early in the next cycle to give more time for testing.